src/lib.rs

```diff
@@ -274,6 +296,23 @@ impl OptionsBuilder {
         self
     }
 
+    /// Will allow the builder to generate an `Options` struct with the provided value.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    ///   use dogstatsd::OptionsBuilder;
+    ///
+    ///   let options_builder = OptionsBuilder::new().batching_options(None);
+    /// ```
+    pub fn batching_options(
+        &mut self,
+        batching_options: Option<BatchingOptions>,
+    ) -> &mut OptionsBuilder {
+        self.batching_options = batching_options;
+        self
+    }
```

Comment:
Since this field will already default to None, I think this builder method can take a non-optional.

```suggestion
    pub fn batching_options(
        &mut self,
        batching_options: BatchingOptions,
    ) -> &mut OptionsBuilder {
        self.batching_options = Some(batching_options);
        self
    }
```